core-api/src/main/java/com/optimizely/ab/odp/ODPEventManager.java

```diff
@@ -111,11 +119,17 @@ private class EventDispatcherThread extends Thread {
 
         private volatile boolean shouldStop = false;
 
+        private final List<ODPEvent> currentBatch = new ArrayList<>();
+
+        private long lastFlushTime = new Date().getTime();
+
         @Override
         public void run() {
             while (true) {
                 try {
-                    ODPEvent nextEvent = eventQueue.poll(FLUSH_INTERVAL, TimeUnit.MILLISECONDS);
+                    long nextFlushMillis = Math.max(0, flushInterval - (new Date().getTime() - lastFlushTime));
+                    ODPEvent nextEvent = eventQueue.poll(nextFlushMillis, TimeUnit.MILLISECONDS);
```

Comment:
Added a check to see if the batch has events, then adding a timeout otherwise doing a no argument poll which indefinitely blocks till the next event arrives.